File: 351002/Matsuyeu/Publisher/src/NewsPortal.Publisher.API/Controllers/v2/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using Publisher.src.NewsPortal.Publisher.Application.Dtos.Auth;
using Publisher.src.NewsPortal.Publisher.Application.Dtos.ResponseTo;
using Publisher.src.NewsPortal.Publisher.Application.Services.Abstractions;
using Publisher.src.NewsPortal.Publisher.Domain.Exceptions;

namespace Publisher.src.NewsPortal.Publisher.API.Controllers.v2;

[ApiController]
[Route("api/v2.0")]
public class AuthController : ControllerBase
{
    private readonly ICreatorService _creatorService;
    private readonly IJwtService _jwtService;
    private readonly ILogger<AuthController> _logger;

    public AuthController(ICreatorService creatorService, IJwtService jwtService, ILogger<AuthController> logger)
    {
        _creatorService = creatorService;
        _jwtService = jwtService;
        _logger = logger;
    }

    [HttpPost("creators")]
    public async Task<ActionResult<CreatorResponseTo>> Register([FromBody] RegisterRequestDto request)
    {
        try
        {
            var creator = await _creatorService.RegisterAsync(request);
            return CreatedAtAction(nameof(GetCreatorById), new { id = creator.Id }, creator);
        }
        catch (ConflictException ex)
        {
            return Conflict(new { errorMessage = ex.Message, errorCode = "40901" });
        }
        catch (Exception ex)
        {
            return BadRequest(new { errorMessage = ex.Message, errorCode = "40001" });
        }
    }

    [HttpGet("creators/{id}")]
    public async Task<ActionResult<CreatorResponseTo>> GetCreatorById(long id)
    {
        var creator = await _creatorService.GetCreatorByIdAsync(id);
        if (creator == null)
            return NotFound(new { errorMessage = "Creator not found", errorCode = "40401" });
        return Ok(creator);
    }

    [HttpPost("login")]
    public async Task<ActionResult<LoginResponseDto>> Login([FromBody] LoginRequestDto request)
    {
        try
        {
            var response = await _creatorService.LoginAsync(request);
            return Ok(response);
        }
        catch (UnauthorizedException)
        {
            return Unauthorized(new { errorMessage = "Invalid login or password", errorCode = "40101" });
        }
    }
}

File: 351002/Bubich/t361/publisher/app/controllers/auth_controller.py
from flask import Blueprint, request, jsonify
from app.services.auth_service import AuthService
from app.exceptions.custom_exceptions import ValidationError

auth_bp = Blueprint('auth', __name__)


class AuthController:
    def __init__(self, auth_service: AuthService):
        self.auth_service = auth_service
        self.register_routes()

    def register_routes(self):
        @auth_bp.route('/writers', methods=['POST'])
        def register():
            """Регистрация нового пользователя: POST /api/v2.0/writers"""
            try:
                data = request.get_json()
                result = self.auth_service.register(data)
                return jsonify(result), 201
            except ValidationError as e:
                return jsonify({
                    "errorMessage": e.message,
                    "errorCode": e.error_code
                }), int(e.error_code[:3])

        @auth_bp.route('/login', methods=['POST'])
        def login():
            """Аутентификация: POST /api/v2.0/login"""
            try:
                data = request.get_json()
                login = data.get('login', '')
                password = data.get('password', '')
                result = self.auth_service.login(login, password)
                return jsonify(result), 200
            except ValidationError as e:
                return jsonify({
                    "errorMessage": e.message,
                    "errorCode": e.error_code
                }), 401